code/sound/sound.cpp

```diff
@@ -904,6 +919,12 @@ sound_handle snd_play_looping(game_snd* gs, float pan, int /*start_loop*/, int /
 	return handle;
 }
 
+void remove_looping_sound(SCP_list<LoopingSoundInfo> &looping_sounds, sound_handle sig)
+{
+	// the cast to void avoids warnings about unused return value
+	(void)std::remove_if(looping_sounds.begin(), looping_sounds.end(), create_ls_info_match_pred(sig));
```

Comment:
Use a lambda here, not a full-up `std::function`.

Reply:
Good point. The intent was to avoid code duplication, but the result ended up making things more complicated, haha.